core/src/main/java/io/micronaut/core/util/localeresolution/FixedLocaleResolver.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.core.util.localeresolution;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ * Generic implementation of {@link io.micronaut.core.util.LocaleResolver} for fixed locale resolution.
+ *
+ * @author Sergio del Amo
+ * @since 2.3.0
+ * @param <T> The context object which will be used to resolve the locale
+ */
+public class FixedLocaleResolver<T> extends DefaultLocaleResolver<T> {
+
+    protected final Locale locale;
+
+    /**
+     *
+     * @param localeResolutionConfiguration Locale Resolution configuration
+     */
+    public FixedLocaleResolver(LocaleResolutionConfiguration localeResolutionConfiguration) {
```

Comment:
if this is a standalone class it doesn't make sense to tie it to configuration. It should simply receive the locale as the argument

Reply:
It needs the configuration to pass it to the parent class.

Reply:
I don't think having the configuration here as a contract makes sense. Fixed does not need a default because its fixed. There is no chance for it to not have a value

Reply:
I've splitted the configuration in two interfaces. Now, I am injecting only the `fixed` configuration in the `fixed` resolver.

Reply:
I've changed FixedLocaleResolver to take a Locale in the constructor.